tests/EvmAcceptanceTests/test/Transfer.ts

```diff
@@ -51,4 +52,113 @@ describe("Transfer ethers", function () {
 
     expect(await ethers.provider.getBalance(payee.address)).to.be.eq(FUND);
   });
+
+  it("should be possible to batch transfer using a smart contract", async function () {
+    const ACCOUNTS_COUNT = 3;
+    const ACCOUNT_VALUE = 1_000_000;
+
+    const accounts = Array.from({length: ACCOUNTS_COUNT}, (v, k) =>
+      ethers.Wallet.createRandom().connect(ethers.provider)
+    );
+
+    const addresses = accounts.map((signer) => signer.address);
+
+    const BatchTransferContract = await ethers.getContractFactory("BatchTransferCtor");
+    const batchTransfer = await BatchTransferContract.deploy(addresses, ACCOUNT_VALUE, {
+      value: ACCOUNTS_COUNT * ACCOUNT_VALUE
+    });
+    await batchTransfer.deployed();
+
+    const balances = await Promise.all(accounts.map((account) => account.getBalance()));
+    balances.forEach((el) => expect(el).to.be.eq(ACCOUNT_VALUE));
+  });
+
+  xit("should be possible to batch transfer using a smart contract with full precision", async function () {
+    const ACCOUNTS_COUNT = 3;
+    const ACCOUNT_VALUE = 1_234_567;
+
+    const accounts = Array.from({length: ACCOUNTS_COUNT}, (v, k) =>
+      ethers.Wallet.createRandom().connect(ethers.provider)
+    );
+
+    const addresses = accounts.map((signer) => signer.address);
+
+    const BatchTransferContract = await ethers.getContractFactory("BatchTransferCtor");
+    const batchTransfer = await BatchTransferContract.deploy(addresses, ACCOUNT_VALUE, {
+      value: ACCOUNTS_COUNT * ACCOUNT_VALUE
+    });
+    await batchTransfer.deployed();
+
+    const balances = await Promise.all(accounts.map((account) => account.getBalance()));
+    balances.forEach((el) => expect(el).to.be.eq(ACCOUNT_VALUE));
+  });
+
+  it("probably should be possible to use sent funds of smart contract", async function () {
+    const TRANSFER_VALUE = 1_000_000;
+
+    // Create random account
+    const rndAccount = ethers.Wallet.createRandom().connect(ethers.provider);
+    const randomAccount = rndAccount.address;
+
+    let initialBal = await ethers.provider.getBalance(randomAccount);
+    expect(initialBal).to.be.eq(0);
+
+    const [owner] = await ethers.getSigners();
+    let InitialOwnerbal = await ethers.provider.getBalance(owner.address);
+
+    // check enough funds + gas
+    expect(InitialOwnerbal).to.be.at.least(TRANSFER_VALUE * 1.1);
+
+    // Deploy with no funds at contract address, but send value
+    const SingleTransferContract = await ethers.getContractFactory("SingleTransfer");
```

Comment:
to have parallelism please do not use ethers.js directly to deploy a contract.

Reply:
Will do.